packages/advisor-components/src/RuleDetails/RuleDetails.tsx

```diff
@@ -0,0 +1,229 @@
+import './RuleDetails.scss';
+
+import React from 'react';
+import { useIntl } from 'react-intl';
+import Markdown from 'react-markdown';
+import rehypeRaw from 'rehype-raw';
+import rehypeSanitize from 'rehype-sanitize';
+
+import { Flex, FlexItem, Stack, StackItem, Text, TextContent, TextVariants } from '@patternfly/react-core';
+import ExternalLinkAltIcon from '@patternfly/react-icons/dist/js/icons/external-link-alt-icon';
+
+import InsightsLabel from '@redhat-cloud-services/frontend-components/InsightsLabel';
+import { SeverityLine } from '@redhat-cloud-services/frontend-components-charts/SeverityLine';
+
+import { IMPACT_LABEL_KEY, LIKELIHOOD_LABEL_KEY, RISK_OF_CHANGE_LABEL_KEY, TOTAL_RISK_LABEL_KEY } from '../constants';
+import RebootRequired from '../RebootRequired/RebootRequired';
+import RuleRating from '../RuleRating/RuleRating';
+import ContextWrapper, { ContextWrapperProps } from '../ContextWrapper';
+import messages from '../messages';
+import { barDividedList, topicLinks } from '../common';
+import { AdvisorProduct, Likelihood, Rating, RiskOfChange, RuleContentOcp, RuleContentRhel, TopicRhel } from '../types';
+
+interface RuleDetailsBaseProps {
+  product: AdvisorProduct;
+  rule: RuleContentOcp | RuleContentRhel;
+  resolutionRisk?: number;
+  resolutionRiskDesc?: string;
+  header?: React.ReactNode;
+  isDetailsPage: boolean;
+  topics: TopicRhel[];
+  /**
+   * onVoteClick - a callback used to update the rating of a particular rule
+   * @param {string} ruleId - ID (usually in plugin|error_key format) of the rule that needs to be updated
+   * @param {number} newRating rating (-1, 0, 1)
+   */
+  onVoteClick?: (ruleId: string, calculatedRating: Rating) => unknown;
+  showViewAffected?: boolean;
+  linkComponent?: any;
```

Comment:
Not sure which type to use here. When I tried `React.Component`, the compiler errored with
`error TS2604: JSX element type 'Link' does not have any construct or call signatures.`

The same error message was obtained after using `React.JSXElementConstructor<LinkProps>` as a type...